src/io/sampling.jl

```diff
@@ -117,20 +159,58 @@ function sample(dom::Domain, n::Int, sampler=SobolSample(); supported_dists=Dict
 
     # Adjust samples for discrete values using flooring trick
     # Ensure unguided scenarios do not have superfluous parameter values
-    return adjust_samples(dom, spec, df)
+    return df
 end
 
+"""
+    sample_site_selection(d::Domain, n::Int, sampler=SobolSample())::DataFrame
+
+Create samples of only site selection parameters and rescale to distribution defined in the model spec.
+
+# Arguments
+- `d` : Domain.
+- `n` : number of samples to generate.
+- `sampler` : type of sampler to use.
+"""
+function sample_site_selection(d::Domain, n::Int, sampler=SobolSample())::DataFrame
+
+    crit_spec = component_params(d.model, Criteria)
+    env_spec = component_params(d.model, EnvironmentalLayer)
+
+    int_spec = component_params(d.model, Intervention)
+    insertcols!(int_spec, :val, :bounds => copy([int_spec[:, :full_bounds]...]))
+    guided_spec = _adjust_guided_lower_bound(int_spec[int_spec[:, :fieldname].==:guided, :], 1)
+    select!(guided_spec, Not(:bounds))
+
+    sample_df = vcat(vcat(env_spec, guided_spec), crit_spec)
+    site_selection_sample = sample(sample_df, n, sampler)
+
+    process_inputs!(sample_df, site_selection_sample)
+    return site_selection_sample
+end
 
 """
     sample_cf(d::Domain, n::Int, sampler=SobolSample())::DataFrame
     
 Generate only counterfactual scenarios using any sampler from QuasiMonteCarlo.jl
 """
 function sample_cf(d::Domain, n::Int, sampler=SobolSample())::DataFrame
-    df::DataFrame = _sample(d, n, sampler)
+    df::DataFrame = sample(d, n, sampler)
     return adjust_cf_samples(d, df)
 end
 
+"""
+    adjust_guided_bounds(guided_spec::DataFrame, lower::Int64)::DataFrame
+    
+Adjust lower bound of guided parameter spec to alter sampling range.
+"""
+function _adjust_guided_lower_bound(spec_df::DataFrame, lower::Int64)::DataFrame
+    guided_col = spec_df.fieldname .== :guided
+    g_upper = spec_df[:, :upper_bound][1]
+    spec_df[guided_col, [:val, :lower_bound, :bounds, :full_bounds]] .= reshape([lower, lower, (lower, g_upper), (lower, g_upper)], (1, 4))
```

Comment:
Streamline this please:

```julia
g_upper = spec_df[:, :upper_bound][1]

# Should be:
spec_df[guided_col, :upper_bound]
```

Reply:
Also just realised this modifies its input (`spec_df`) so please add `!` to the function name

Reply:
It was throwing a size mismatch error because LHS was (1,4) and RHS was (4,)

Reply:
Yes - so column data (vector) was being provided to update a row.

No need to complicate things with an additional call, just provide the row:

```julia
julia> df = DataFrame(rand(3,5), :auto)
3×5 DataFrame
 Row │ x1          x2        x3        x4           x5       
     │ Float64     Float64   Float64   Float64      Float64  
─────┼───────────────────────────────────────────────────────
   1 │ 0.00723953  0.202598  0.318617  0.000524996  0.554819
   2 │ 0.687003    0.640692  0.812998  0.351252     0.960522
   3 │ 0.393727    0.37577   0.435917  0.119245     0.825794

julia> df[3, :] .= [1; 2; 3; 4; 5]
DataFrameRow
 Row │ x1       x2       x3       x4       x5      
     │ Float64  Float64  Float64  Float64  Float64 
─────┼─────────────────────────────────────────────
   3 │     1.0      2.0      3.0      4.0      5.0

# OR
julia> df[3, :] .= [1 2 3 4 5]'
DataFrameRow
 Row │ x1       x2       x3       x4       x5      
     │ Float64  Float64  Float64  Float64  Float64 
─────┼─────────────────────────────────────────────
   3 │     1.0      2.0      3.0      4.0      5.0

julia> df
3×5 DataFrame
 Row │ x1          x2        x3        x4           x5       
     │ Float64     Float64   Float64   Float64      Float64  
─────┼───────────────────────────────────────────────────────
   1 │ 0.00723953  0.202598  0.318617  0.000524996  0.554819
   2 │ 0.687003    0.640692  0.812998  0.351252     0.960522
   3 │ 1.0         2.0       3.0       4.0          5.0
```

Reply:
Yeah I tried the semicolons but it didn't change the size (maybe cause of the Tuples insie?) 
![size_output](https://user-images.githubusercontent.com/56939532/218925688-5647fce3-8b3c-40d5-98b7-1e3c3426303a.PNG)

The transpose seems to change the size but then when I try the allocation it throws an error, something about how the adjoint doesn't exist for vectors of tuples...

Reply:
As above, use semi-colons instead of commas:

`df[3, :] .= [1; 2; 3; 4; 5]`

Or remove the commas and transpose:

`df[3, :] .= [1 2 3 4 5]'`

Reply:
As in the screenshot, the semicolons don't seem to change the size and the transpose errors
![transpose_error](https://user-images.githubusercontent.com/56939532/218927559-a56e922b-705b-4c40-8d80-baeb088b5496.PNG)

Reply:
can you see the screen shots?

Reply:
The screen shots show you're using commas?

Reply:
in line 3 I use semicolons

Reply:
see these 2 lines give the same size
![size_output2](https://user-images.githubusercontent.com/56939532/218928635-65a54d44-0149-471f-83e8-4470cdcb6aae.png)

Reply:
![image](https://user-images.githubusercontent.com/4075136/218928746-194073b9-4bce-43e0-81eb-74d9c302ca01.png)

Reply:
![dim_error_semicols](https://user-images.githubusercontent.com/56939532/218929665-9985e4db-e1b6-4b1f-b6de-42408ab2c62f.PNG)

Reply:
![image](https://user-images.githubusercontent.com/4075136/218930624-49d83f69-da64-445f-b460-f26812cae176.png)

Reply:
oh so you don't need commas at all! My comma version was erroring as well

Reply:
![image](https://user-images.githubusercontent.com/4075136/218931148-dad45914-e48b-4e07-a34c-8be472f90aca.png)

Have you kept your environment up to date?

Reply:
Also, can you check `guided_col`? It should select a single row.

Reply:
guided_col is a one-element BitVector

Reply:
it seems like it was g_upper that was causing the error, it was a vector of one element for some reason

Reply:
updating everything now just incase

Reply:
works now by making `g_upper = spec_df[guided_col, :upper_bound][1]`